### PR TITLE
node 0.12.7, 0.8.26 + npm upgrade to 2.14.4 for all node versions

### PR DESCRIPTION
**Pull request by chrmoritz**
This PR moves node 0.12.7 to versions, upgrades node08 to 0.8.26 and upgrades npm in all node formulas (which are build with npm) to 2.14.4 to ensure compatibility with newer node versions.

Refs: https://github.com/Homebrew/homebrew/pull/43973

### CODE REVIEW

File: node012.rb [hunk 1]
```diff
@@ -0,0 +1,147 @@
+class Node < Formula
+  desc "Platform built on the V8 JavaScript runtime to build network applications"
+  homepage "https://nodejs.org/"
+  url "https://nodejs.org/dist/v0.12.7/node-v0.12.7.tar.gz"
+  sha256 "b23d64df051c9c969b0c583f802d5d71de342e53067127a5061415be7e12f39d"
+  head "https://github.com/nodejs/node.git", :branch => "v0.12"
+  revision 1
+
+  bottle do
```
Comment (DomT4) [inline node012.rb]: Cut this block. We'll add a new one.

File: node012.rb [hunk 2]
```diff
@@ -0,0 +1,147 @@
+class Node < Formula
+  desc "Platform built on the V8 JavaScript runtime to build network applications"
+  homepage "https://nodejs.org/"
+  url "https://nodejs.org/dist/v0.12.7/node-v0.12.7.tar.gz"
+  sha256 "b23d64df051c9c969b0c583f802d5d71de342e53067127a5061415be7e12f39d"
+  head "https://github.com/nodejs/node.git", :branch => "v0.12"
+  revision 1
```
Comment (DomT4) [inline node012.rb]: Cut this line.

File: node012.rb [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    sha256 "15689cc474a79975eaa6d791b24e6fa021494839c9b691ac307d74acefc5f834" => :yosemite
+    sha256 "a7a7d37c6e5088ed3f58b867d4d246851715d3a4f2f3b4b3c40cc7452ff6728c" => :mavericks
+    sha256 "374f3c5b576e4173590b8413e9941df121a84f46bd48161fc758e1f7d42e0402" => :mountain_lion
+  end
+
+  option "with-debug", "Build with debugger hooks"
+  option "without-npm", "npm will not be installed"
+  option "without-completion", "npm bash completion will not be installed"
+
+  deprecated_option "enable-debug" => "with-debug"
```
Comment (DomT4) [inline node012.rb]: Kill this line.

File: node012.rb [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+  depends_on :python => :build if MacOS.version <= :snow_leopard
+  depends_on "pkg-config" => :build
+  depends_on "openssl" => :optional
+
+  # https://github.com/nodejs/node-v0.x-archive/issues/7919
+  # https://github.com/Homebrew/homebrew/issues/36681
+  depends_on "icu4c" => :optional
```
Comment (DomT4) [inline node012.rb]: Add a `conflicts_with "node", :because => "Differing version of same formula"` line one empty line underneath this.

**Comment by DomT4** [comment]
Few nits but :+1:

File: node012.rb [hunk 5]
```diff
@@ -0,0 +1,139 @@
+class Node012 < Formula
+  desc "Platform built on the V8 JavaScript runtime to build network applications"
```
Comment (DomT4) [inline node012.rb]:
```
homebrew/versions/node012:
 * Description is too long. "name: desc" should be less than 80 characters.
Length is calculated as homebrew/versions/node012 + desc. (currently 100)
```

File: node08.rb [hunk 6]
```diff
@@ -27,8 +26,8 @@ class Node08 < Formula
   end
 
   resource "npm" do
-    url "https://registry.npmjs.org/npm/-/npm-2.12.1.tgz"
-    sha256 "6b6512c6f9097da193dfe046053d6d0483b5c5658dc0a763c1ba5609b6bbc16c"
+    url "https://registry.npmjs.org/npm/-/npm-2.14.4.tgz"
+    sha256 "c8b602de5d51f956aa8f9c34d89be38b2df3b7c25ff6588030eb8224b070db27"
   end
 
   # Fixes double-free issue. See https://github.com/joyent/node/issues/6427
```
Comment (DomT4) [inline node08.rb]: This seems to be redundant/broken now:

```
==> Verifying node08-0.8.28.tar.gz checksum
tar xf /Library/Caches/Homebrew/node08-0.8.28.tar.gz
==> Patching
patching file deps/v8/src/spaces.h
Hunk #1 FAILED at 321.
1 out of 1 hunk FAILED -- saving rejects to file deps/v8/src/spaces.h.rej
Error: Failure while executing: /usr/bin/patch -g 0 -f -p1
```

**Comment by DomT4** [comment]
Couple more nits, but nearly there on this.

File: node08.rb (hunk 6 above)
Comment (chrmoritz) [inline node08.rb]: It's fixed since 0.8.27, removing the patch.

**Comment by chrmoritz** [comment]
Looks like Jenkins had some issues checking out this PR.

**Comment by chrmoritz** [comment]
@DomT4: it's green now

**Comment by DomT4** [comment]
I noticed. I've been watching the bot for the last two hours :wink:. Bear with me just a second.

**Comment by DomT4** [comment]
Merged. Thank you for getting this done @chrmoritz :beer:

**Pull request closed without merging**